File: sort/selectionSort.js
function selectionSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let min = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[min] > arr[j]) {
        min = j;
      }
    }
    let temp = arr[min];
    arr[min] = arr[i];
    arr[i] = temp;
  }
  return arr;
}

console.log(
  selectionSort([2, 2, 312, 4, 345, 34, 1, 12, 34, 534, 123, 142, 1])
);

File: python/selectionSort.py
# O(n*n)
def selectionSort(arr):
    size = len(arr)

    for i in range(size):
        minIdx = i

        #find min val
        for j in range(i + 1, size):
            if arr[j] < arr[minIdx]:
                minIdx = j #swap the minIdx to i

        #set minVal to currect index
        (arr[i], arr[minIdx]) = (arr[minIdx], arr[i])

data = [-2, 45, 0, 11, -9]

print(data)
selectionSort(data)
print(data)


